pandas/tests/arrays/string_/test_string.py

```diff
@@ -194,6 +194,12 @@ def test_constructor_raises():
     with pytest.raises(ValueError, match="sequence of strings"):
         pd.arrays.StringArray(np.array([]))
 
+    with pytest.raises(ValueError, match="strings or pandas.NA"):
```

Comment:
Can you add a test for pd.NaT as well? I guess could get in there if passing a bunch of ISO strings